samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -881,4 +919,23 @@ void BLEConnectivityManager::Recovery::StartTimer()
 	k_timer_start(&mRecoveryTimer, K_SECONDS(time), K_NO_WAIT);
 }
 
+BLEConnectivityManager::State BLEConnectivityManager::GetCurrentState()
+{
+	for (uint8_t s = State::Scanning; s <= State::Idle; s++) {
+		if (mStateBitmask & (1 << s)) {
+			return static_cast<State>(s);
+		}
+	}
+	return State::Unknown;
+}
+
+void BLEConnectivityManager::UpdateStateFlag(State newState, bool enabled)
+{
+	uint8_t newStateBitmask = enabled ? mStateBitmask | (1 << newState) : mStateBitmask & ~(1 << newState);
+	if (newStateBitmask != mStateBitmask && mStateChanged) {
+		mStateBitmask = newStateBitmask;
```

Comment:
I guess we should update the `mStateChanged` regardless of the callback being nullptr or not.

Reply:
I guess, you meant `mStateBitmask` instead of `mStateChanged` update. :)

Reply:
Right :)